build/nuke/Build.Steps.cs

```diff
@@ -163,6 +163,15 @@ partial class Build
                 .EnableNoRestore()
                 .SetFramework(TargetFramework.NETCOREAPP3_1)
                 .SetOutput(TracerHomeDirectory / TargetFramework.NETCOREAPP3_1));
+
+            DotNetPublish(s => s
```

Comment:
note: one version of the same assembly is already embedded to native.dll (not sure if worth separating?)

Reply:
I'm not saying it's directly issue with this PR, just pointing out on the way some comments. There is no better way than direct marks in the code. 

All I'm saying is that thanks to StartupHook, the loader is only necessary for Windows net461. Currently it's binary is shipped embedded to the dll + separately in netcoreapp3.1 deploy (so it's just a question of optimized shipping). Linux and Mac deploy could actually remove it completely simplifying native build, but then StartupHook would need some refactoring. 

At first I thought StartupHook is going to be the main way loading, because net461 is just the only exceptional case.